### ISSUE TITLE
Scssify filter can result in BOM in HTML

### ISSUE THREAD

**Issue by benburwell** [jekyll/jekyll#3914]
I'm using the `scssify` filter to spit out a `:compressed` stylesheet in a `<style>` block. Since my SCSS file contains non-ASCII characters, [Sass places a byte-order mark at the beginning of the compressed output](https://github.com/sass/sass/issues/1449). So effectively, I have:

```
<style>

[BOM]a{color:red}

</style>
```

Having the BOM immediately before a selector appears to be an issue, as that rule is not interpreted in Chrome. Changing the Sass mode to `:compact` replaces the BOM with a `@charset` rule, which works fine. I'm not sure if this would be an issue with Sass or with Jekyll (or with Chrome, as FF appears to ignore the BOM), but I figured I'd look for some feedback here.

FWIW, the [W3C CSS Validator](https://jigsaw.w3.org/css-validator/#validate_by_input) also chokes up on BOM occurring after beginning-of-input:

```
    ﻿a { color: red; }
```

(there is a BOM just before the `a`, so copy-and-paste carefully)

From what I've read, the CSS spec requires implementations to interpret the BOM at the beginning of the file, but I haven't seen anything about it occurring in the middle.

**Comment by envygeeks** [comment]
This isn't a bug in Chrome, the opposite would be a bug in FF IMO because ignoring mistakes and proceeding is whacky behavior, but it's not something that is even important.  At minimum I think this would be a bug in SASS since we pass it onto them to deal with AFAIW we could work around it in Jekyll though I don't know if that's worth it, all things considered, I don't think we heavily lock SASS so if they rolled out a fix in theory you could get it as fast as they rolled it out as long as it's not major.

**Comment by benburwell** [comment]
@envygeeks, I'm totally with you on the browser issue. I'm not too familiar with the internals of Jekyll, so please correct me if I'm wrong. It seems that the `scssify` filter and the SCSS converter both (reasonably) use the same Sass plugin.

I was looking through the Sass source and I noticed a distinction between the `compile` and `compile_file` functions. Perhaps adjusting `compile` to not insert the BOM but leaving it in `compile_file` would be the most sensible thing. I'm not sure whether that would work well within the Sass architecture, but I'll see what they have to say.

My next question is, would this actually result in the desired behavior in Jekyll? It looks to me like `scssify` uses the same converter that compiling a SCSS file does. Would this need to be changed as well?

**Comment by envygeeks** [comment]
I think it would have to change, I'd have to go back and double check but if it can't change we can always strip the BOM as it's easily detectable, we could even re-encode to remove it as part of that larger problem we were having cross-platform internationally anyways (there was a ticket, I'll search for it this afternoon and link it to this one) and if we did that enforcement and normalization that was discussed it would probably solve this problem but I don't think that was ever marked for Jekyll 3 (@parkr would have to confirm that) and if it wasn't I don't think we could get it done before the deadline that was set for the 3.0 release but for sure I could get in the BOM stripping before the deadline because that's no biggie.

**Comment by benburwell** [comment]
I'd be happy to submit a PR for BOM stripping if you think that's the way to go? Seems like the simplest solution, but I'll defer to people who really know the architecture.

**Comment by envygeeks** [comment]
That would be great!

**Comment by parkr** [comment]
Is this Windows-only?

**Comment by benburwell** [comment]
I'm running Jekyll on Ubuntu 15.04.

**Comment by pathawks** [comment]
Is this fixed in [**jekyll-sass-converter** 1.4.0](https://github.com/jekyll/jekyll-sass-converter/releases/tag/v1.4.0)?

**Comment by parkr** [comment]
Yes!
